cloudbridge/cloud/factory.py

```diff
@@ -12,6 +16,54 @@
 log = logging.getLogger(__name__)
 
 
+# Todo: Move to pyeventsystem if we're keeping this logic
+class ParentMiddlewareManager(SimpleMiddlewareManager):
+
+    def __init__(self, event_manager=None):
+        super(ParentMiddlewareManager, self).__init__(event_manager)
+        self.middleware_constructors = []
+
+    def add_constructor(self, middleware_class, *args):
```

Comment:
This could possibly be renamed to add_middleware_class, accepting the class in question, and both the constructor args and kwargs.